tests/helper/helper_cmd_wrapper.go

```diff
@@ -30,11 +36,74 @@ func Cmd(program string, args ...string) *CmdWrapper {
 	}
 }
 
-func (cw *CmdWrapper) ShouldPass() *CmdWrapper {
+func (cw *CmdWrapper) Runner() *CmdWrapper {
 	fmt.Fprintln(GinkgoWriter, runningCmd(cw.Cmd))
-	session, err := gexec.Start(cw.Cmd, cw.writer, cw.writer)
-	Expect(err).NotTo(HaveOccurred())
-	cw.session = session
+	cw.session, cw.err = gexec.Start(cw.Cmd, cw.writer, cw.writer)
+	timeout := time.After(cw.timeout)
+	if cw.timeout > 0 {
+		select {
+		case <-cw.stopChan:
+			if cw.session != nil {
+				if runtime.GOOS == "windows" {
+					cw.session.Kill()
+				} else {
+					cw.session.Terminate()
+				}
+			}
+		case <-timeout:
+			if cw.session != nil {
+				if runtime.GOOS == "windows" {
+					cw.session.Kill()
+				} else {
+					cw.session.Terminate()
+				}
+			}
+		}
+	}
+	return cw
+}
+
+func (cw *CmdWrapper) WithRetry(maxRetry int, intervalSeconds time.Duration) string {
+	cw.maxRetry = maxRetry
+	for i := 0; i < cw.maxRetry; i++ {
+		fmt.Fprintf(GinkgoWriter, "try %d of %d\n", i, cw.maxRetry)
+		cw.Runner()
+		cw.session.Wait()
+		// if exit code is 0 which means the program succeeded and hence we retry
```

Comment:
Why should we retry if the command was successful? Shouldn’t it be other way around? If it fails then we retry

Reply:
That function is now updated, it will work with both shouldpass and shouldfail now.